src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -186,11 +202,76 @@ public DynamoView(DynamoViewModel dynamoViewModel)
                  }
              };
 
+            // Add an event when the collection is modified.   
+            tabItems.CollectionChanged += this.OnCollectionChanged;
```

Comment:
you need to unsubscribe this when the dynamoView is disposed or closed. Checkout the dispose method of this class.

Reply:
yeah. I will add that.